motionplan/motionPlanner.go

```diff
@@ -381,3 +382,81 @@ IK:
 	}
 	return orderedSolutions, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
+func CheckPlan(
+	checkFrame frame.Frame,
```

Comment:
What is the `checkFrame`?

Reply:
it is the frame we build the solverFrame from

Reply:
From the caller's perspective, what is the check frame? How do I as a caller of this function know which frame to provide as the checkFrame?

Reply:
it is easiest for me to explain this in terms of an example.
In `builtin_test.go` the last subtest we have is called `"ensure transforms of obstacles works - collision with base"`
In this subtest we ensure that we have a collision with the base.
We do so by supplying `kinBase.Kinematics()` as the value for checkFrame
------
Now, suppose instead of supplying `kinBase.Kinematics()` as the value for checkFrame I supplied the value with `cameraFrame`
Now, the function would return a nil error since there is no collision.
----
perhaps one way to think about this is as such. I want the frame I passed in the be checked for collisions as well as check all the frame's children for collisions.

Reply:
Got it, we are asking if that frame or any of it's children will collide with anything